### PR TITLE
Commit PR notes to draftlogs and add scripts to provide draft release notes (CHANGELOG.md) and empty draftlogs

### LINKED ISSUE #5778: Commit PR logs and add a script to help draft the next release notes
Each PR that adds/changes/fixes something notable in terms of CHANGELOG, should provide a Markdown file named using PR number (or the issue number) and the category of change in a folder.
For example a file could be created unders `draftlogs/5678_fix.md` for a bug fix or `draftlogs/5678_add.md` for a new feature, etc.
Having these files on master at any time one could simply npm run script to draft the next release notes.
Also we could use this script to quickly draft, then finalize the changelog.
This folder could automatically be emptied during the release process to help collect the changes for the next release.

### PR DESCRIPTION
**Pull request by archmoj**
Resolves #5778.

Running
```sh
npm run use-draftlogs
``` 
simply updates `CHANGELOG.md`.

@plotly/plotly_js

### CODE REVIEW

**Comment by nicolaskruchten** [comment]
I like it!

File: draftlogs/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (alexcjohnson) [inline draftlogs/.gitignore:1]: Did you mean this file to be `.gitkeep`? I guess there's a [competing convention to use `.gitignore`](https://stackoverflow.com/questions/7229885/what-are-the-differences-between-gitignore-and-gitkeep) but I agree with the sentiment that this is more confusing than `.gitkeep`.

Comment (archmoj) [reply draftlogs/.gitignore:1]: Yes. I could change that if you want.

File: tasks/use_draftlogs.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        all.Added.push(message);
+    } else if(filename.endsWith('_remove.md')) {
+        all.Removed.push(message);
+    } else if(filename.endsWith('_deprecate.md')) {
+        all.Deprecated.push(message);
+    } else if(filename.endsWith('_change.md')) {
+        all.Changed.push(message);
+    } else if(filename.endsWith('_fix.md')) {
+        all.Fixed.push(message);
+    }
```
Comment (alexcjohnson) [inline tasks/use_draftlogs.js:52]: Else throw an error, logging the filename as having been ignored.

Might want to also loosen the `endsWith` tests - to eg `filename.toLowerCase().indexOf('_add') !== -1` so stuff like `5678_Added.md` will still be picked up.

**Comment by alexcjohnson** [comment]
Looks good! Can you add notes on creating these md files to the PR template, and add notes on the new package scripts to the release procedure?

File: tasks/use_draftlogs.js (hunk 2 above)
Comment (archmoj) [reply tasks/use_draftlogs.js:52]: I think we have a test which disallows use of uppercase in filenames.

Comment (alexcjohnson) [reply tasks/use_draftlogs.js:52]: We [do test for uppercase](https://github.com/plotly/plotly.js/blob/7374eb8efd9ff9fa7b38ec24b0c1b5e9f953fd70/tasks/test_syntax.js#L210-L216), but I don't think it'll dig into the `draftlogs` dir. And there's still add vs adds vs added etc. Anyway the main thing is to throw an error if we ignore the file. After adding that if you still want to be strict about names that's ok, it'll just occasionally make more work during release if we get an incorrect name.

File: draftlogs/.gitignore (hunk 1 above)
Comment (alexcjohnson) [reply draftlogs/.gitignore:1]: Either .gitkeep, or a README explaining these files - actually a README might be better, then the PR template could just refer to it, but when contributors are actually writing these files they'll have the README to look at.

Comment (archmoj) [reply draftlogs/.gitignore]: Good call. Done in https://github.com/plotly/plotly.js/pull/5780/commits/3a91c0e380886d6292b12a71b22fd25d298b5f82.

File: tasks/use_draftlogs.js (hunk 2 above)
Comment (archmoj) [reply tasks/use_draftlogs.js:55]: Good call. Addressed in https://github.com/plotly/plotly.js/pull/5780/commits/3a91c0e380886d6292b12a71b22fd25d298b5f82.

**Comment by archmoj** [comment]
> Looks good! Can you add notes on creating these md files to the PR template, and add notes on the new package scripts to the release procedure?

Good call. Done in https://github.com/plotly/plotly.js/pull/5780/commits/eefab9b8178f40c60b138bb7544ef13886d1ec34.

File: draftlogs/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+## Directory of draft logs to help prepare the upcoming [CHANGELOG](https://github.com/plotly/plotly.js/blob/master/CHANGELOG.md)
+
+It is required that the PR contributors start the filename with a number.
+This number should preferably be the PR number.
+The number of issue they are trying to close could also be used.
+The markdown file should end with one of the followings:
```
Comment (alexcjohnson) [inline draftlogs/README.md:6]:
```suggestion
Every pull request should add at least one markdown file to this directory.
The filename must start with a number, preferably the PR number, followed by one of these:
```

File: draftlogs/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+1. `_fix.md` to propose a bug fix
+2. `_add.md` to propose new features
+3. `_remove.md` to propose a feature removal
+4. `_change.md` to propose a minor/major change
+5. `_deprecate.md` to propose a feature deprecate
```
Comment (alexcjohnson) [inline draftlogs/README.md:11]:
```suggestion
5. `_deprecate.md` to propose a feature be deprecated

If your PR falls into more than one category - for example adding a new feature and changing an existing feature - you should include each in a separate file.
```

**Review by alexcjohnson — APPROVED** [review]
Looks good! 💃

**Pull request merged**